### DISCUSSION TITLE
[8.0 QA] Removed react from peer dependency

### DISCUSSION THREAD

**Discussion by vanessayuenn** [storybookjs/storybook#25685] [Announcement]
#### Feature owner: @JReinhold 
Please comment in this discussion thread for any bug report and feedback for this feature as you test out Storybook 8. Kindly adhere to the following format:

> **Project**: Name of your project as entered in Notion
> **Version**: The version of storybook you tested with
> **Bug/feedback**: Details of the bugs/feedback you have

Feature owner should triage the comments here frequently and open issues as needed.

**Comment by JReinhold** [comment]
**Project**: SvelteUI
**Version**: 8.0.0-alpha.13
**Bug/feedback**:

Running `upgrade` or `automigrate` didn't trigger the "Remove React dependency" automigration. This is in a monorepo, but `react` and `react-dom` is listed in `package.json` in CWD (the root).

**Reply by valentinpalkovic** [reply]
Issue:

The condition that checks whether the automigration should run or not doesn't take prereleases into account:

```js
if (!semver.gte(storybookVersion, '8.0.0')) {
  return null;
}
```

> If a version has a prerelease tag (for example, 1.2.3-alpha.3) then it will only be allowed to satisfy comparator sets if at least one comparator with the same [major, minor, patch] tuple also has a prerelease tag.
> 
> For example, the range >1.2.3-alpha.3 would be allowed to match the version 1.2.3-alpha.7, but it would not be satisfied by 3.4.5-alpha.9, even though 3.4.5-alpha.9 is technically "greater than" 1.2.3-alpha.3 according to the SemVer sort rules. The version range only accepts prerelease tags on the 1.2.3 version. The version 3.4.5 would satisfy the range, because it does not have a prerelease flag, and 3.4.5 is greater than 1.2.3-alpha.7.
> 
> The purpose for this behavior is twofold. First, prerelease versions frequently are updated very quickly, and contain many breaking changes that are (by the author's design) not yet fit for public consumption. Therefore, by default, they are excluded from range matching semantics.
> 
> Second, a user who has opted into using a prerelease version has clearly indicated the intent to use that specific set of alpha/beta/rc versions. By including a prerelease tag in the range, the user is indicating that they are aware of the risk. However, it is still not appropriate to assume that they have opted into taking a similar risk on the next set of prerelease versions.
> 
> Note that this behavior can be suppressed (treating all prerelease versions as if they were normal versions, for the purpose of range matching) by setting the includePrerelease flag on the options object to any [functions](https://github.com/npm/node-semver#functions) that do range matching.

Source: https://www.npmjs.com/package/semver#prerelease-tags

**Reply by valentinpalkovic** [reply]
Well, false alarm! `storybookVersion` is already coerced and reduced from `8.0.0-alpha.13` to `8.0.0`. The migration wasn't added to the index.ts file and was, therefore, never considered.

**Reply by valentinpalkovic** [reply]
https://github.com/storybookjs/storybook/pull/25717

**Comment by shilman** [comment]
**Project** Storyblok
**Version** 8.0.0-alpha.13

Yarn1 warnings after removing `react`/`react-dom`:

```
warning "@storybook/addon-essentials > @storybook/addon-backgrounds > @storybook/icons@1.2.3" has unmet peer dependency "react@^16.8.0 || ^17.0.0 || ^18.0.0".
warning "@storybook/addon-essentials > @storybook/addon-backgrounds > @storybook/icons@1.2.3" has unmet peer dependency "react-dom@^16.8.0 || ^17.0.0 || ^18.0.0".
warning "@storybook/addon-essentials > @storybook/addon-controls > @storybook/blocks@8.0.0-alpha.13" has unmet peer dependency "react@^16.8.0 || ^17.0.0 || ^18.0.0".
warning "@storybook/addon-essentials > @storybook/addon-controls > @storybook/blocks@8.0.0-alpha.13" has unmet peer dependency "react-dom@^16.8.0 || ^17.0.0 || ^18.0.0".
warning "@storybook/addon-essentials > @storybook/addon-docs > @mdx-js/react@3.0.0" has unmet peer dependency "@types/react@>=16".
warning " > @storybook/theming@8.0.0-alpha.13" has unmet peer dependency "react@^16.8.0 || ^17.0.0 || ^18.0.0".
warning " > @storybook/theming@8.0.0-alpha.13" has unmet peer dependency "react-dom@^16.8.0 || ^17.0.0 || ^18.0.0".
warning "@storybook/addon-essentials > @storybook/addon-controls > @storybook/blocks > markdown-to-jsx@7.3.2" has unmet peer dependency "react@>= 0.14.0".
warning "@storybook/addon-essentials > @storybook/addon-controls > @storybook/blocks > react-colorful@5.6.1" has unmet peer dependency "react@>=16.8.0".
warning "@storybook/addon-essentials > @storybook/addon-controls > @storybook/blocks > react-colorful@5.6.1" has unmet peer dependency "react-dom@>=16.8.0".
warning " > @testing-library/vue@7.0.0" has unmet peer dependency "@vue/compiler-sfc@>= 3".
warning " > @vue/vue3-jest@29.2.6" has unmet peer dependency "jest@29.x".
warning " > postcss-scss@4.0.9" has unmet peer dependency "postcss@^8.4.29".
warning "stylelint-config-recommended-vue > stylelint-config-recommended@14.0.0" has incorrect peer dependency "stylelint@^16.0.0".
warning " > stylelint-config-sass-guidelines@10.0.0" has unmet peer dependency "postcss@^8.4.21".
```

**Reply by vanessayuenn** [reply]
sev: S2

**Reply by JReinhold** [reply]
https://github.com/storybookjs/storybook/pull/25926 fixes this for all warnings reported by Yarn 2+, not those only shown in Yarn 1

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.4
**Bug/feedback:**

When loading any MDX (docs only or autodocs) in Storybook 8.0.0-alpha-4 or higher, there is a warning in the console mentioning that emotion is loaded multiple times:

```
You are loading @emotion/react when it is already loaded. Running multiple instances may cause problems. This can happen if multiple versions are used, or if multiple builds of the same version are used.
```

Apart from emotion being instantiated by the Manager UI, there are two moments where it gets instantiated in the Preview UI:

Via `@storybook/components`
<img width="512" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/dba03fde-4baf-4065-8808-aa5e541da954">

Via `@storybook/addon-docs/node_modules/@storybook/components`
<img width="768" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/ae724b41-3637-4b63-9a9c-1119505acdf4">



**Repro:**
1. Go to https://github.com/yannbf/mealdrop/tree/bug/multiple-instances-emotion
2. Run Storybook
3. Check the logs

**Reply by vanessayuenn** [reply]
tracked in https://github.com/storybookjs/storybook/issues/25761

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

Since Storybook 8.0.0-alpha.4, any autodocs stories in MealDrop won't work anymore, because the `theme` in one of the Storybook components is undefined:
<img width="724" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/8b3f57bc-4ed1-4a4e-87ff-374a1a01b268">


Leading into this issue:
![image](https://github.com/storybookjs/storybook/assets/1671563/21d4070c-3ea8-4b3b-a1d5-80c20f358ef1)

The problem goes away when downgrading to 8.0.0-alpha.3. Might be related to the [multiple emotion instances issue](https://github.com/storybookjs/storybook/issues/25761).

**Repro:**
1. Go to https://github.com/yannbf/mealdrop/tree/bug/autodocs-issue
2. Visit any story that uses autodocs (all stories but the Button)
